supafunc/errors.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing_extensions import TypedDict
+
+
+class FunctionsError(Exception):
+    def __init__(self, message: str) -> None:
+        Exception.__init__(self, message)
+        self.message = message
+        self.name = "FunctionsError"
+
+
+class FunctionsApiErrorDict(TypedDict):
+    name: str
+    message: str
+    status: int
+
+
+class CustomFunctionsError(FunctionsError):
```

Comment:
I think we can get rid of `FunctionsError`, the only thing it is doing that CustomFunctionsError doesn't do is set `self.message` (I'd just add that line to `CustomFunctionsError.__init__` , remove FunctionsError and then rename CustomFunctionsError to FunctionsError)

Also do we need to store the name as an attribute? It seems to be just the error class name

Reply:
The `name` attribute was mainly to conform to the likes of the `gotrue-py` library's error class. I don't think we have standardise on what information the error class should be holding.